exchanges/binance.js

```diff
@@ -0,0 +1,593 @@
+      quantity: amount,
+      price: price,
+    },
+    _.bind(setOrder, this)
+  );
+};
+
+Trader.prototype.getOrder = function(order, callback) {
+  var get = function(err, data) {
+    log.debug(
+      '[binance.js] entering "getOrder" callback after api call, err:',
+      err,
+      ' data:',
+      data
+    );
+
+    if (!err && !_.isEmpty(data.msg)) err = new Error(data.msg);
+
+    if (err)
+      return log.error(
+        '[binance.js] unable to get order',
+        order,
+        JSON.stringify(err)
+      );
+
+    var price = parseFloat(data.price);
+    var amount = parseFloat(data.executedQty);
+    var date = moment.unix(data.time);
+
+    callback(undefined, { price, amount, date });
+  }.bind(this);
+
+  this.binance.queryOrder(
+    {
+      symbol: this.pair,
+      orderId: order,
+    },
+    get
+  );
+};
+
+Trader.prototype.buy = function(amount, price, callback) {
+  this.addOrder('buy', amount, price, callback);
+};
+
+Trader.prototype.sell = function(amount, price, callback) {
+  this.addOrder('sell', amount, price, callback);
+};
+
+Trader.prototype.checkOrder = function(order, callback) {
+  var check = function(err, data) {
+    log.debug(
+      '[binance.js] entering "checkOrder" callback after api call, err:',
+      err,
+      ' data:',
+      data
+    );
+
+    if (!err && !_.isEmpty(data.msg)) err = new Error(data.msg);
+
+    if (err)
+      return log.error(
+        '[binance.js] Unable to check order',
+        order,
+        JSON.stringify(err)
+      );
+
+    var stillThere =
+      data.status === 'NEW' || data.status === 'PARTIALLY_FILLED';
+    callback(err.message, !stillThere);
+  };
+
+  this.binance.queryOrder(
+    {
+      symbol: this.pair,
+      orderId: order,
+    },
+    _.bind(check, this)
+  );
+};
+
+Trader.prototype.cancelOrder = function(order, callback) {
+  var args = _.toArray(arguments);
+  var cancel = function(err, data) {
+    log.debug(
+      '[binance.js] entering "cancelOrder" callback after api call, err:',
+      err,
+      ' data:',
+      data
+    );
+
+    if (!err && !_.isEmpty(data.msg)) err = new Error(data.msg);
+
+    if (err) {
+      log.error(
+        '[binance.js] unable to cancel order',
+        order,
+        '(',
+        err,
+        JSON.stringify(err),
+        ')'
+      );
+      return this.retry(this.cancelOrder, args, err);
+    }
+
+    callback();
+  };
+
+  this.binance.cancelOrder(
+    {
+      symbol: this.pair,
+      orderId: order,
+    },
+    _.bind(cancel, this)
+  );
+};
+
+Trader.getCapabilities = function() {
+  return {
+    name: 'Binance',
+    slug: 'binance',
+    currencies: ['BTC', 'BNB', 'ETH', 'USDT'],
+    assets: [
+      'BTC',
+      'BCC',
+      'BCG',
+      'BNB',
+      'DASH',
+      'ETH',
+      'ETC',
+      'EOS',
+      'NEO',
+      'OMG',
+      'POWR',
+      'QTUM',
+      'ZEC',
+    ],
+    markets: [
```

Comment:
FYI: this is fine, but I am okay with skipping prettier (using `// prettier-ignore`) in these kind of config blocks.

Reply:
Noted, I didn't realize that prettier did this to me :)